eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -12,21 +12,41 @@ parameters:
   default: []
 - name: JobTemplatePath
   type: string
+# Set this to false to do a full checkout for private repositories with the azure pipelines service connection
```

Comment:
I think we would need to figure out how to look at the repo name and do a template condition so we don't have to have different code checked into our public and private repos.

Reply:
It's easy enough to do a repo name check to just skip the sparse checkout, but I'd like to figure out a way to load creds for the pr repos in that case.